packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,717 @@
+      ...(props.record.protocols && { protocols: props.record.protocols }),
+      ...(props.record.tags && { tags: props.record.tags }),
+    })
+
+    if (props.record.data) {
+      const newModel = props.record.data
+      newModel.client_certificate = newModel.client_certificate?.id
+      updateModel(props.record.data)
+    } else if (props.record.config) {
+      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
+        updateModel({
+          service_id: props.record.service_id,
+          route_id: props.record.route_id,
+          consumer_id: props.record.consumer_id,
+          consumer_group_id: props.record.consumer_group_id,
+          enabled: props.record.enabled,
+        })
+      }
+
+      updateModel(props.record.config, 'config')
+    }
+  }
+
+  // Check if incoming field exists in current model and add if so update
+  if (entityIdField.value && props.entityId && props.schema) {
+    const updateFields: Record<string, any> = {}
+    const key = entityIdField.value === 'consumer_group_id' ? 'consumer_group-id' : JSON.parse(JSON.stringify(entityIdField.value).replace('_', '-'))
+
+    // ex. set consumer-id: <entityId>
+    if (Object.prototype.hasOwnProperty.call(formModel, key)) {
+      updateFields[key] = props.entityId
+    }
+
+    updateModel(updateFields)
+  }
+
+  // Check if entity field exists in current model and if so update
+  // credentials don't recognize field with -id, so for now set it like
+  // ex. consumer: <entityId>
+  // we'll fix this on submit
+  if (props.entityId && props.schema && props.isCredential) {
+    const updateFields: Record<string, any> = {}
+    if (Object.prototype.hasOwnProperty.call(formModel, props.entityType)) {
+      updateFields[props.entityType] = props.entityId
+    }
+
+    updateModel(updateFields)
+  }
+
+  loading.value = false
+}
+
+watch(loading, (newLoading) => {
+  emit('loading', newLoading)
+})
+
+watch(() => props.schema, (newSchema) => {
+  const form: Record<string, any> = parseSchema(newSchema)
+
+  Object.assign(formModel, form.model)
+
+  formSchema.value = { fields: formSchema.value?.fields?.map((r: Record<string, any>) => { return { ...r, disabled: r.disabled || false } }) }
+  Object.assign(originalModel, JSON.parse(JSON.stringify(form.model)))
+  sharedFormName.value = getSharedFormName(form.model.name)
+
+  initFormModel()
+}, { immediate: true })
+
+onBeforeMount(() => {
+  form.value = parseSchema(props.schema)
+
+  Object.assign(formModel, form.value?.model || {})
+  formSchema.value = form.value?.schema || {}
+
+  initFormModel()
+})
+</script>
+
+<style lang="scss" scoped>
+.kong-ui-entities-plugin-form {
+  width: 100%;
+
+  .fade-enter-active,
+  .fade-leave-active {
+    transition: opacity 0.5s;
+  }
+
+  .fade-enter-from, .fade-leave-to {
+    opacity: 0;
+  }
+}
+</style>
+
+<style lang="scss">
+.kong-ui-entities-plugin-form {
+  .vue-form-generator {
```

Comment:
suggestion: instead of using unscoped styles here, could you use the `:deep` selector to target the child component(s)?